### PR TITLE
cephadm: add test coverage for NFSGanesha, HAproxy, & Keepalived classes

### PR DESCRIPTION
**Pull request by phlogistonjohn**
(Part of) Fixes: https://tracker.ceph.com/issues/57621

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [x] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [x] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

File: src/cephadm/tests/test_nfs.py [hunk 1]
```diff
@@ -0,0 +1,564 @@
...
+            [
+                SAMPLE_UUID,
+                "wilma",
+                haproxy_json(files=False),
+                SAMPLE_HAPROXY_IMAGE,
+            ]
+        ),
+    ],
+)
+def test_haproxy_validation_errors(args):
```
Comment (adk3798) [inline src/cephadm/tests/test_nfs.py:283]: any particular reason for also including the haproxy and keepalive tests here? Just thinking down the line if I wanted to look at a test for haproxy or keepalived I don't think I would intuitively think to check in a `test_nfs` file.

**Review by adk3798 — APPROVED** [review]
LGTM

File: src/cephadm/tests/test_nfs.py [hunk 2]
```diff
@@ -0,0 +1,564 @@
...
+                "keepalived.conf": "",
+            },
+        }
+    return {}
+
+
+@pytest.mark.parametrize(
+    "args,kwargs",
+    [
+        (["foobar", "fred", good_nfs_json()], {}),
```
[the remark this answers is not in the document]
Comment (phlogistonjohn) [reply src/cephadm/tests/test_nfs.py:75]: Sure.

File: src/cephadm/tests/test_nfs.py (hunk 1 above)
Comment (phlogistonjohn) [reply src/cephadm/tests/test_nfs.py]: Done.

**Comment by phlogistonjohn** [comment]
Rebased.

File: src/cephadm/tests/test_nfs.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (phlogistonjohn) [reply src/cephadm/tests/test_nfs.py:44]: Fixed.

**Comment by phlogistonjohn** [comment]
I think this is ready for another round of reviews (@adk3798, et al.)

**Review by adk3798 — APPROVED** [review]
LGTM

**Pull request merged**